app/scenes/Settings/Security.tsx

```diff
@@ -38,22 +42,74 @@ function Security() {
     [showToast, t]
   );
 
+  const saveData = React.useCallback(
+    async (newData) => {
+      try {
+        setData(newData);
+        await auth.updateTeam(newData);
+        showSuccessMessage();
+      } catch (err) {
+        showToast(err.message, {
+          type: "error",
+        });
+      }
+    },
+    [auth, showSuccessMessage, showToast]
+  );
+
   const handleChange = React.useCallback(
     async (ev: React.ChangeEvent<HTMLInputElement>) => {
       const newData = { ...data, [ev.target.id]: ev.target.checked };
-      setData(newData);
-      await auth.updateTeam(newData);
-      showSuccessMessage();
+      await saveData(newData);
     },
-    [auth, data, showSuccessMessage]
+    [data, saveData]
   );
 
-  const handleDefaultRoleChange = async (newDefaultRole: string) => {
-    const newData = { ...data, defaultUserRole: newDefaultRole };
-    setData(newData);
-    await auth.updateTeam(newData);
-    showSuccessMessage();
-  };
+  const handleDefaultRoleChange = React.useCallback(
+    async (newDefaultRole: string) => {
+      const newData = { ...data, defaultUserRole: newDefaultRole };
+      await saveData(newData);
+    },
+    [data, saveData]
+  );
+
+  const handleAllowSignupsChange = React.useCallback(
+    async (ev: React.ChangeEvent<HTMLInputElement>) => {
+      const inviteRequired = !ev.target.checked;
+      const newData = { ...data, inviteRequired };
+
+      if (inviteRequired) {
+        dialogs.openModal({
+          isCentered: true,
+          title: t("Are you sure you want to disable authorized signups?"),
+          content: (
+            <DisableSignupsConfirmation
+              onSubmit={async () => {
+                await saveData(newData);
+              }}
+              submitText={t("I’m sure — Disable")}
+              savingText={`${t("Disabling…")}`}
```

Comment:
```suggestion
              savingText={`${t("Disabling")}…`}
```

We've been keeping these out of the translation string for consistency